dotnet/src/SemanticKernel/KernelConfig.cs

```diff
@@ -273,6 +282,35 @@ public KernelConfig SetDefaultTextEmbeddingGenerationService(string serviceId)
         return this;
     }
 
+    /// <summary>
+    /// Set REST API operation runner.
+    /// </summary>
+    /// <remarks>
+    /// This method should be used when the same handlers(logging, resiliency, etc), used by HttpClient, are applicable to all hosts REST API operations are run against.
+    /// </remarks>
+    /// <param name="httpClient">An instance of the HttpClient class.</param>
+    /// <param name="authCallback">Optional callback for adding auth data to the API requests.</param>
+    /// <returns>The updated kernel configuration.</returns>
+    public KernelConfig SetRestApiOperationRunner(HttpClient httpClient, AuthenticateRequestAsyncCallback? authCallback = null)
```

Comment:
could you refactor this code to use the same approach used for AI services? instead of passing instances, which are expensive, register a factory.